lib/puppet/provider/mount.rb

```diff
@@ -26,21 +26,33 @@ def mount
   def remount
     #TRANSLATORS refers to remounting a file system
     info _("Remounting")
-    if resource[:remounts] == :true
+    os = Facter.value(:operatingsystem)
+    supports_remounts = (resource[:remounts] == :true)
+    if supports_remounts && os == 'AIX'
+      remount_with_option("remount")
+    elsif os.match(/^(FreeBSD|DragonFly|OpenBSD)$/)
+      remount_with_option("update")
+    elsif supports_remounts
       mountcmd "-o", "remount", resource[:name]
-    elsif ["FreeBSD", "DragonFly", "OpenBSD"].include?(Facter.value(:operatingsystem))
-      if self.options && !self.options.empty?
-        options = self.options + ",update"
-      else
-        options = "update"
-      end
-      mountcmd "-o", options, resource[:name]
     else
       unmount
       mount
     end
   end
 
+  def remount_with_option(option)
```

Comment:
This method would benefit from documentation indicating that it'll both remount with a specific option, as well as the contents of the `options` parameter if the field is set.

Reply:
agreed. update pushed